horizons/util/checkupdates.py

```diff
@@ -19,99 +19,102 @@
 # 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 # ###################################################
 
+import json
 import platform
-import socket
+import threading
 import urllib.error
-import urllib.parse
 import urllib.request
 import webbrowser
 
 from fife.extensions.pychan.widgets import Button
 
+import horizons.main
 from horizons.constants import NETWORK, VERSION
+from horizons.extscheduler import ExtScheduler
 from horizons.gui.windows import Popup
 from horizons.i18n import gettext as T
+from horizons.util.python.callback import Callback
 
 
-class UpdateInfo:
-	INVALID, READY, UNINITIALIZED = list(range(3))
-	def __init__(self):
-		self.status = UpdateInfo.UNINITIALIZED
-		self.version = None # type: str
-		self.link = None # type: str
-
 TIMEOUT = 5.0 # we should be done before the user can start a game
-def check_for_updates(info):
-	"""Check if there's a new version.
-	@return update file contents or None"""
-	# make sure to always set info.status, but only when we're done
+
+
+def check_for_updates():
+	"""
+	Check if there's a new version, returns the data from the server, otherwise None.
+	"""
 	# no updates for git version
 	if VERSION.IS_DEV_VERSION:
-		info.status = UpdateInfo.INVALID
 		return
 
-	# only updates for operating systems missing a packagemanagement
-	if (platform.system() == 'Windows' or platform.system() == 'Darwin'):
-		# retrieve current version w.r.t. the local version.
-		# this way, possible configurations of different most recent versions should be handleable in the future.
-		data = urllib.parse.urlencode( {"my_version" : VERSION.RELEASE_VERSION} )
-		url = NETWORK.UPDATE_FILE_URL
-		try:
-			u = urllib.request.urlopen( url + "?" + data, timeout=TIMEOUT )
-		except (urllib.error.URLError, socket.timeout):
-			# Silently ignore the failed update, printing stuff might crash the game
-			# if no console is available
-			info.status = UpdateInfo.INVALID
-			return
-
-		version = u.readline()
-		link = u.readline()
-		u.close()
-
-		version = version[:-1] # remove newlines
-		link = link[:-1] # remove newlines
-
-		if version != VERSION.RELEASE_VERSION:
-			# there is a new version
-			info.version = version
-			info.link = link
-			info.status = UpdateInfo.READY
-		else:
-			info.status = UpdateInfo.INVALID
-	else:
-		info.status = UpdateInfo.INVALID
+	# only updates for operating systems missing a package management
```

Comment:
Interesting idea for sure.

One potential issue for people running from git, that I see, is that it might be a developer, and if we were to check for new commits (maybe tags?) you'd get the popup very often. But perhaps you had something else in mind?